### ISSUE TITLE
File/path name length error

### ISSUE THREAD

**Issue by schuylermartin45** [prefix-dev/rattler-build#778]
Somewhere between versions `0.13.0` and `0.14.2` of `rattler-build`, I've started getting an error message about a file name/path being too long.

```
 ╭─ Running build for recipe: conda-recipe-manager-0.1.0-pyh4616a5c_0
 │
 │ ╭─ Fetching source code
 │ │ Fetching source from path: "/home/runner/work/conda-recipe-manager/conda-recipe-manager"
 │ │
 │ ╰─────────────────── (took 1 second)
 │
 ╰─────────────────── (took 1 second)
 × error Error building package: IO Error: File name too long (os error 36)
Error:   × IO Error: File name too long (os error 36)
  ╰─▶ File name too long (os error 36)
```

Breaking Example (v0.14.2)
https://github.com/conda-incubator/conda-recipe-manager/actions/runs/8603702384/job/23576152730?pr=22

Working Example (v0.13.0) (note that `/home/runner/work/conda-recipe-manager/conda-recipe-manager` is referenced in the logs and the build continues)
https://github.com/conda-incubator/conda-recipe-manager/actions/runs/8438404927/job/23111180963

**Comment by wolfv** [comment]
Weird! Do you happen to have the recipe that is tested around? Can look into this tomorrow :)

**Comment by schuylermartin45** [comment]
It's the recipe for this project, converted from the `meta.yaml` file by the script. I'll get you as close to the output as I can this morning.

**Comment by wolfv** [comment]
@schuylermartin45 I think you are executing the build from inside the directory. Currnetly the output dir is also set to inside the direcotry.

Maybe you could try something like `--output-dir=$HOME/output` and see if that helps? Would be interesting.

**Comment by schuylermartin45** [comment]
Here are some more experiments I've done locally on my Mac, in case they are interesting, all performed inside my top-level `conda-recipe-manager` directory

1)
```sh
rattler-build build -r recipe --output-dir ~/temp
```
`rattler-build` errors out with an OS error, I'm guessing it can't interpret the `~` and maybe that's a quality of life improvement to look into later for users.

2)
```sh
rattler-build build -r recipe --output-dir $HOME/temp
```
Works as expected

3)
```sh
rattler-build build -r recipe
```
Works as example 2, dumping the build to `conda-recipe-manager/output`

For S's and G's, I'll change the output directory for the GitHub workflow and see if that makes a difference shortly.

**Comment by schuylermartin45** [comment]
The workflow now passes:
https://github.com/conda-incubator/conda-recipe-manager/actions/runs/8621362556/job/23630037413?pr=22

I just made a few changes:
```sh
mkdir -p ../temp
rattler-build build -r recipe/ --output-dir=../temp
```
Building in the same directory does seem to work on some platforms and had been working fine in previous versions of `rattler-build`. I can live with this change and I'll leave it up to you if you'd like to investigate further or close this issue.

**Comment by wolfv** [comment]
This is basically a duplicate of #576

**Comment by lorepirri** [comment]
Hi there,

I experienced the same issue as @schuylermartin45 ... meaning I was running `rattler-build` from within the same folder of the recipe and I had random errors (symlink creation and the here mentioned `File name too long (os error 36)`) for now to me clear reasons.

Since I lost a little bit of time before realizing this (I did see the temp folders `output` and ` pytest-of-root` in the recipe but did not connect the dots immediately), to save time for other fellow builders, it would be nice to have a warning or better terminate the build command with an error because it's a nondeterministic situation anyway: it will fail most of the time.

While I can't re-open this issue, I could open a new one if you think it is worth investing some time in this.
